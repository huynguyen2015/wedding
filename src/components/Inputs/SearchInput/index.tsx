import React, { useEffect, useState } from 'react'
import { Input } from 'antd'
import { usePrevious } from '../../Hooks/usePrevious'
import isEqual from 'lodash/isEqual'

interface SearchInputProps {
  value?: number | string
  defaultValue?: number | string
  onPressEnter?: (value) => void
  onChange?: (value) => void
  onBlur?: (value) => void
  prefix?: React.ReactNode
  disabled?: boolean
  placeholder?: string
  small?: boolean
}

const SearchInput: React.FC<SearchInputProps> = ({
  value,
  defaultValue,
  onPressEnter,
  onChange,
  onBlur,
  prefix,
  disabled,
  placeholder,
  small
}) => {
  const previousValue = usePrevious(value)
  const [textValue, setTextValue] = useState(value)

  useEffect(() => {
    if (!isEqual(previousValue, value)) {
      setTextValue(value)
    }
  }, [previousValue, value])

  const triggerChange = () => {
    if (onChange) {
      onChange(textValue)
    }
  }

  const onTextChange = (e) => {
    setTextValue(e.target?.value)
  }

  const handleBlur = () => {
    triggerChange()
    if (onBlur) {
      onBlur(textValue)
    }
  }

  const handlePressEnter = (e) => {
    triggerChange()
    if (onPressEnter) {
      onPressEnter(e.target?.value)
    }
  }

  return (
    <Input
      value={textValue}
      className="full-width"
      defaultValue={defaultValue}
      onChange={onTextChange}
      onBlur={handleBlur}
      onPressEnter={handlePressEnter}
      disabled={disabled}
      placeholder={placeholder || ''}
      size={small ? 'small' : 'large'}
      prefix={prefix}
    />
  )
}

export default SearchInput
