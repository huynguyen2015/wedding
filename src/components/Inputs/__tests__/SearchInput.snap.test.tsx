import SearchInput from '../SearchInput/index'
import renderer from 'react-test-renderer'

describe('SearchInput', function () {
  let dataTable
  beforeEach(function () {})

  afterEach(function () {
    dataTable = null
  })

  it('To match snapshot search input', async () => {
    dataTable = renderer
      .create(
        <SearchInput
          value={1}
          onPressEnter={() => {}}
          onChange={() => {}}
          onBlur={() => {}}
          placeholder="Currency input"
          small={false}></SearchInput>
      )
      .toJSON()

    expect(dataTable).toMatchSnapshot()
  })
})
