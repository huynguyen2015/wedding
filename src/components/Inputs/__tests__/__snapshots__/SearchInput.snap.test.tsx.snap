// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput To match snapshot search input 1`] = `
<input
  className="ant-input ant-input-lg full-width"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  placeholder="Currency input"
  type="text"
  value="1"
/>
`;
